examples/iris.rb

```diff
@@ -0,0 +1,103 @@
+require 'numo/narray'
+require 'chainer'
+require "datasets"
+
+class IrisChain < Chainer::Chain
+  L = Chainer::Links::Connection::Linear
+  F = Chainer::Functions
+
+  def initialize(n_units, n_out)
+    super()
+    init_scope do
+      @l1 = L.new(nil, out_size: n_units)
+      @l2 = L.new(nil, out_size: n_out)
+    end
+  end
+
+  def call(x, y)
+    return F::Loss::MeanSquaredError.mean_squared_error(fwd(x), y)
+  end
+
+  def fwd(x)
+    h1 = F::Activation::Sigmoid.sigmoid(@l1.(x))
+    h2 = @l2.(h1)
+    return h2
+  end
+end
+
+model = IrisChain.new(6,3)
+
+optimizer = Chainer::Optimizers::Adam.new
+optimizer.setup(model)
+
+iris = Datasets::Iris.new
+x = iris.each.map {|r| r.each.to_a[0..3]}
+
+# target
+y_class = iris.each.map {|r| r.class}
+
+# class index array
+# ["Iris-setosa", "Iris-versicolor", "Iris-virginica"]
+class_name = y_class.uniq
+# y => [0, 0, 0, 0, ,,, 1, 1, ,,, ,2, 2]
+y = y_class.map{|s|
+  class_name.index(s)
+}
+
+# y_onehot => One-hot [[1.0, 0.0, 0.0], [1.0, 0.0, 0.0],,, [0.0, 1.0, 0.0], ,, [0.0, 0.0, 1.0]]
+y_onehot = y_class.map{|s|
+  i = class_name.index(s)
+  a = Array.new(class_name.length, 0.0)
```

Comment:
`size`, `length` are mixed.
I would like to unite `length`

Reply:
Thank you for pointing out.
Fixed with 1f2a607.

I unified it to use the "size" method according to other processing.